p2p/sync/sync.go

```diff
@@ -336,10 +336,15 @@ func (s *Service) adaptAndSanityCheckBlock(
 
 			newClasses := make(map[felt.Felt]core.Class)
 			for _, cls := range classes {
-				coreC := p2p2core.AdaptClass(cls)
+				coreC, err := p2p2core.AdaptClass(cls)
+				if err != nil {
+					bodyCh <- BlockBody{Err: fmt.Errorf("unsupported class: %w", err)}
```

Comment:
```suggestion
					bodyCh <- BlockBody{Err: fmt.Errorf("failed to adapt class: %w", err)}
```
because it's not necessary unsupported class, but can be wrong format or something else.